test/functional/normalization/fluentd_structured_configuration_option2.txt

```diff
@@ -0,0 +1,454 @@
+      remove_keys log,stream,CONTAINER_ID_FULL,CONTAINER_NAME
+      process_kubernetes_events 'true'
+    </formatter>
+    <formatter>
+      tag "kubernetes.var.log.containers**"
+      type k8s_json_file
+      remove_keys log,stream,CONTAINER_ID_FULL,CONTAINER_NAME
+    </formatter>
+    <elasticsearch_index_name>
+      enabled 'true'
+      tag "journal.system** system.var.log** **_default_** **_kube-*_** **_openshift-*_** **_openshift_**"
+      name_type static
+      static_index_name infra-write
+    </elasticsearch_index_name>
+    <elasticsearch_index_name>
+      enabled 'true'
+      tag "linux-audit.log** k8s-audit.log** openshift-audit.log**"
+      name_type static
+      static_index_name audit-write
+    </elasticsearch_index_name>
+    <elasticsearch_index_name>
+      enabled 'true'
+      tag "**"
+      name_type static
+      static_index_name app-write
+    </elasticsearch_index_name>
+  </filter>
+
+  <filter **>
+    @type elasticsearch_genid_ext
+    hash_id_key viaq_msg_id
+    alt_key kubernetes.event.metadata.uid
+    alt_tags 'kubernetes.var.log.containers.logging-eventrouter-*.** kubernetes.var.log.containers.eventrouter-*.** kubernetes.var.log.containers.cluster-logging-eventrouter-*.** kubernetes.journal.container._default_.kubernetes-event'
+  </filter>
+
+  #flatten labels to prevent field explosion in ES
+  <filter ** >
+    @type record_transformer
+    enable_ruby true
+    <record>
+      kubernetes ${!record['kubernetes'].nil? ? record['kubernetes'].merge({"flat_labels": (record['kubernetes']['labels']||{}).map{|k,v| "#{k}=#{v}"}}) : {} }
+    </record>
+    remove_keys $.kubernetes.labels
+  </filter>
+
+  # Relabel specific source tags to specific intermediary labels for copy processing
+  # Earlier matchers remove logs so they don't fall through to later ones.
+  # A log source matcher may be null if no pipeline wants that type of log.
+  <match **_default_** **_kube-*_** **_openshift-*_** **_openshift_** journal.** system.var.log**>
+    @type null
+  </match>
+  <match kubernetes.**>
+    @type relabel
+    @label @_APPLICATION
+  </match>
+  <match linux-audit.log** k8s-audit.log** openshift-audit.log**>
+    @type null
+  </match>
+
+  <match **>
+    @type stdout
+  </match>
+
+</label>
+
+# Relabel specific sources (e.g. logs.apps) to multiple pipelines
+<label @_APPLICATION>
+  <match **>
+    @type copy
+
+    <store>
+      @type relabel
+      @label @FORWARD_PIPELINE
+    </store>
+  </match>
+</label>
+
+
+# Relabel specific pipelines to multiple, outputs (e.g. ES, kafka stores)
+<label @FORWARD_PIPELINE>
+  <filter **>
+    @type record_transformer
+    <record>
+      openshift { "labels": {} }
+    </record>
+  </filter>
+  <match **>
+    @type copy
+
+    <store>
+      @type relabel
+      @label @FLUENTFORWARD
+    </store>
+  </match>
+</label>
+
+# Ship logs to specific outputs
+<label @FLUENTFORWARD>
+
+  ##### THIS IS NEW CONFIGURATION THAT WILL BE SET USING API
```

Comment:
We already have an embedded plugin which will parse JSON.  Additionally, this filter already exists to accomplish the same thing https://docs.fluentd.org/filter/parser